### PR TITLE
Crazy Visual Stuff Fix + New App Icon

### LINKED ISSUE #5: crazy visual stuff in list view when returning to stories feed
user report:

```
Incase they have trouble with the bug in their app that I mentioned here is a little more data:

It occurs regularly (but not always)  when returning to the story list after reading a story. 

Here is a screen shot of sort of what it looks like when it happens (all photos go away and the screen flickers) in the 2nd screen you can see  sort of the middle of a flicker. 
```

ios_crazy_artifact
ios_crazy_artifact2

### PR DESCRIPTION
**Pull request by ghost**
Fixes #5 (_hopefully!_)

### CODE REVIEW

**Comment by nilnilnil** [comment]
@quanganhdo thoughts?

**Comment by ghost** [comment]
This PR also now fixes #7 

I've also changed the _Deployment Target_ to iOS 7 after our discussion on Skype, and fixed all but one of the deprecation warnings. The one o/s warning is related to a 3rd party lib that's included as a submodule.

File: DuckDuckGo/Constants.m [hunk 1]
```diff
@@ -16,9 +16,11 @@
 
 NSString * const kDDGCustomStoriesURLString = @"https://duckduckgo.com/news.js?o=json&t=m&q=";
 
-NSString * const kDDGTypeInfoURLString = @"https://hunter.duckduckgo.com/watrcoolr.js?o=json&type_info=1";
-NSString * const kDDGStoriesURLString = @"https://hunter.duckduckgo.com/watrcoolr.js?o=json&s=";
-//NSString * const kDDGTypeInfoURLString = @"https://watrcoolr.duckduckgo.com/watrcoolr.js?o=json&type_info=1";
-//NSString * const kDDGStoriesURLString = @"https://watrcoolr.duckduckgo.com/watrcoolr.js?o=json&s=";
+//NSString * const kDDGTypeInfoURLString = @"https://hunter.duckduckgo.com/watrcoolr.js?o=json&type_info=1";
```
Comment (nilnilnil) [inline DuckDuckGo/Constants.m:19]: Just remove please.

File: DuckDuckGo/DDGAppDelegate.m [hunk 2]
```diff
@@ -42,12 +42,15 @@ - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wdeprecated-declarations"
     //Use deprecated uniqueIdentifier call for debug purposes only.
-    [TestFlight setDeviceIdentifier:[[UIDevice currentDevice] uniqueIdentifier]];
+    //[TestFlight setDeviceIdentifier:[[UIDevice currentDevice] uniqueIdentifier]];
```
Comment (nilnilnil) [inline DuckDuckGo/DDGAppDelegate.m:45]: Any reason to keep?

**Comment by nilnilnil** [comment]
Can you give an overview of what you think the problem is and how you solved it?

File: DuckDuckGo/DDGSettingsViewController.m [hunk 3]
```diff
@@ -147,8 +147,9 @@ -(void)configure {
     
     NSString *homeViewMode = [defaults objectForKey:DDGSettingHomeView];
     [self addRadioOptionWithTitle:@"Stories" value:DDGSettingHomeViewTypeStories key:DDGSettingHomeView selected:[homeViewMode isEqual:DDGSettingHomeViewTypeStories]];
-    [self addRadioOptionWithTitle:@"Recents" value:DDGSettingHomeViewTypeRecents key:DDGSettingHomeView selected:[homeViewMode isEqual:DDGSettingHomeViewTypeRecents]];
+    [self addRadioOptionWithTitle:@"Recent" value:DDGSettingHomeViewTypeRecents key:DDGSettingHomeView selected:[homeViewMode isEqual:DDGSettingHomeViewTypeRecents]];
```
Comment (nilnilnil) [inline DuckDuckGo/DDGSettingsViewController.m:150]: thx.

**Comment by ghost** [comment]
The problem, as far as I could tell, was two-fold.

1 - Each time the view was displayed, the stories were refreshed. If you did this in quick succession, the HTTP requests and subsequent UI refreshes would get backed up.
2 - Each time a cell was being rendered, the blurred image was being recreated from scratch rather than using the cached one. This operation is memory and CPU intensive.

For 1) I've added a minimum time interval between story refreshes. It's low, but enough to prevent the HTTP requests from backing up. For 2) I've used the cached image (_so blurred images are only ever created once_) and rewrote the cell to be far more performant.

**Comment by nilnilnil** [comment]
:+1:

**Pull request merged**